include/hipSYCL/sycl/libkernel/generic/hiplike/group_functions.hpp

```diff
@@ -536,6 +535,111 @@ T group_inclusive_scan(Group g, T x, BinaryOperation binary_op) {
   return binary_op(scratch[wid - 1], local_x);
 }
 
+// shift_left
+template <typename Group, typename T>
+HIPSYCL_KERNEL_TARGET
+T shift_group_left(Group g, T x, typename Group::linear_id_type delta = 1) {
+  __shared__   std::aligned_storage_t<sizeof(T) * 1024, alignof(T)> scratch_storage;
```

Comment:
Would it make sense to have a `inline constexpr size_t max_group_size = 1024` somewhere?

Reply:
Yes, that makes a lot of sense. Do you have any preferences, where you want this line?

Reply:
If you want it just for CUDA or HIP, put in the respective file in a corresponding namespace. Don't have preferences apart from that :)